### PR TITLE
Gareth enhance sort

### LINKED ISSUE #45: As a forgetful user, I can sort the list in either decreasing or increasing order
...so that I can quickly sift through the list in any desired direction to find the contact I am looking for.
This is an enhancement to #34, by providing the user the option to sort either by name or by module code in either increasing (A-Z) or decreasing (Z-A) order.

### PR DESCRIPTION
**Pull request by GarethOng**
*Sort Function Enhancements*
- SortLexicographicalCommand shortened to SortCommand
- SortCommand now takes in arguments (preamble, name and moduleCode)
- New utils class: Order to specify the order in which the List is sorted in.

*New way to call sort command for user*
- sort A-Z n/name => sort in increasing order by name
- sort Z-A n/name => sort in decreasing order by name
- sort A-Z m/module => sort in increasing order by moduleCode
- sort Z-A m/module => sort in decreasing order by moduleCode

Any suggestions for easier command names?

### CODE REVIEW

**Review by normkoh — COMMENTED** [review]
LGTM! I think the current sort command naming pattern is pretty good already. I feel that it is short and intuitive. I can't think of anything better.

**Comment by nitant-p** [comment]
LGTM! I think the command words are short and simple. Should be clear to the user 👍

File: src/test/java/seedu/address/logic/commands/StudentCommandTest.java [hunk 1]
```diff
@@ -151,7 +152,7 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         }
 
         @Override
-        public void sortContactLexicographical() {
+        public void sort(Order order, boolean hasName, boolean hasModuleCode) {
             throw new AssertionError("This method should not be called.");
```
Comment (Jonaspng) [inline src/test/java/seedu/address/logic/commands/StudentCommandTest.java:161]: Maybe you could give a more detailed error message on what went wrong?

**Review by Jonaspng — APPROVED** [review]
LGTM.

**Review by JJoeYi — COMMENTED** [review]
LGTM. Naming is intuitive and works for me!

File: src/test/java/seedu/address/logic/commands/StudentCommandTest.java (hunk 1 above)
Comment (GarethOng) [reply src/test/java/seedu/address/logic/commands/StudentCommandTest.java:161]: Noted with thanks.

**Pull request merged**